shared/src/model.ts

```diff
@@ -33,7 +33,7 @@ export class BaseModel {
 
     get isNew() {
         const id = this[(this.constructor as any).idField]
-        return isEmpty(id) || id === NEW_ID
+        return isNil(id) || id === NEW_ID
```

Comment:
Based on the shared spec failures I don't think I can just use the prior `isNil` version, I'll look at fixing it some other way.

Reply:
hm, it seems like we do want isEmpty, right?  I'd switched to that since ID is `string | number` and we're sometimes initializing it to `''`

<img width="185" alt="image" src="https://user-images.githubusercontent.com/79566/117678844-63e25780-b175-11eb-8ab9-0f8e840c69ff.png">

Reply:
Ug I didn't realize that 😭 

Maybe we shouldn't use type checks and just do: `id != NEW_ID && id != ''`

Reply:
hm, actually maybe we should still check isNil tho, so:

`!isNil(id) && id != NEW_ID && id != ''`

Reply:
Agree testing the inverse makes sense, I'll add some tests to confirm the behavior and have you take a peek.